shoop/core/cache.py

```diff
@@ -0,0 +1,161 @@
+
+    def get_version(self, cache_key):
+        """
+        Get the cache version (or None) for the given cache key/namespace.
+
+        The cache version is stored in thread-local storage for the current request, so unless
+        bumped in-request, all `get`s within a single request should get coherently versioned
+        data from the cache.
+
+        :param cache_key: Cache key or namespace
+        :type cache_key: str
+        :return: Version ID or none
+        :rtype: str|none
+        """
+        namespace = _get_cache_key_namespace(cache_key)
+        if not hasattr(_versions, namespace):
+            version = self._cache.get("_version:%s" % namespace)
+            setattr(_versions, namespace, version)
+        else:
+            version = getattr(_versions, namespace, None)
+        return version
+
+    def set(self, key, value, timeout=None, version=None):
+        """
+        Set the value for key `key` in the cache.
+
+        Unlike `django.core.caches[using].set()`, this also derives a timeout
+        and versioning information from the key (and cached version data) if the
+        key begins with a colon-separated namespace, such as `foo:bar`.
+
+        :param key: Cache key
+        :type key: str
+        :param value: Value to cache
+        :type value: object
+        :param timeout: Timeout seconds or None (for auto-determination)
+        :type timeout: int|None
+        :param version: Version string or None (for auto-determination)
+        :type version: str|None
+        :param using: Cache alias
+        :type using: str
+        """
+        if timeout is None:
+            timeout = get_cache_duration(key)
+        if version is None:
+            version = self.get_version(key)
+        self._cache.set(key, value, timeout=timeout, version=version)
+
+    def get(self, key, version=None, default=None):
+        """
+        Get the value for key `key` in the cache.
+
+        Unlike `django.core.caches[using].get()`, versioning information
+        can be auto-derived from the key (and cached version data) if the
+        key begins with a colon-separated namespace, such as `foo:bar`.
+
+        :param key: Cache key
+        :type key: str
+        :param version: Version string or None (for auto-determination)
+        :type version: str|None
+        :param default: Default value, if the key is not found
+        :type default: object
+        :param using: Cache alias
+        :type using: str
+        :return: cached value
+        :rtype: object
+        """
+        if version is None:
+            version = self.get_version(key)
+        return self._cache.get(key, default=default, version=version)
+
+    def clear(self):
+        self._cache.clear()
+
+
+_default_cache = VersionedCache("default")
+get = _default_cache.get
+set = _default_cache.set
+bump_version = _default_cache.bump_version
+clear = _default_cache.clear
+
+# TODO: Add API for PersistentCacheEntry?
```

Comment:
And PersistentCacheEntry should be deprecated...

Can we just rename the PersistentCacheEntry to ConfigurationEntry so that migrations would move the previous installation key to the new configuration API?